src/crowbar.mli

```diff
@@ -1,52 +1,128 @@
-type state
-
-type 'a printer = Format.formatter -> 'a -> unit
+(** {1:top Types } *)
 
 type 'a gen
+(** ['a gen] knows how to generate ['a] for use in Crowbar tests. *)
 
 type ('k, 'res) gens =
   | [] : ('res, 'res) gens
   | (::) : 'a gen * ('k, 'res) gens -> ('a -> 'k, 'res) gens
+(** multiple generators are provided to functions using a listlike syntax.
+    for example, [map [int; int] (fun a b -> a + b)] *)
+
+type 'a printer = Format.formatter -> 'a -> unit
+(** pretty-printers for items generated by Crowbar; useful for the user in
+    translating test failures into bugfixes. *)
+
+(**/**)
 (* re-export stdlib's list
    We only want to override [] syntax in the argument to Map *)
 type nonrec +'a list = 'a list = [] | (::) of 'a * 'a list
+(**/**)
+
+(** {1:generators Generators } *)
+
+(** {2:simple_generators Simple Generators } *)
+
+val int : int gen
+(** [int] generates an integer.  For 32-bit systems, it will generate an
+    Int32.t; otherwise, it will generate an Int64.t .  If you need integers from
+    a smaller domain, consider using {!range}. *)
+
+val uint8 : int gen
+(** [uint8] generates an unsigned byte, ranging from 0 to 255 inclusive. *)
+
+val int8 : int gen
+(** [int8] generates a signed byte, ranging from -128 to 127 inclusive. *)
+
+val int32 : Int32.t gen
+(** [int32] generates a 32-bit signed integer. *)
+
+val int64 : Int64.t gen
+(** [int64] generates a 64-bit signed integer. *)
+
+val float : float gen
+(** [float] generates a double-precision floating-point number. *)
+
+val bytes : string gen
+(** [bytes] generates a string of arbitrary length (including zero-length strings).  *)
+
+val bool : bool gen
+(** [bool] generates a yes or no answer. *)
+
+val range : int -> int gen
+(** [range n] is a generator for integers between 0 (inclusive) and [n] (exclusive). *)
```

Comment:
Range should fail quickly with Invalid_argument if `n <= 0`, and this should be mentioned here. (I think currently it fails, but only when the generator runs and it should fail earlier).

Reply:
range now does what I'd like the docs to say it does (3f6d2b33d9fd562e81d3ed9c9274835f5274f0a2)